### PR TITLE
[CAPI] tiny issues fixings

### PR DESCRIPTION
**Pull request by riverlijunjie**
### Details:
 - ov_infer_request_wait_for return value
 - documentation fixing

### Tickets:
 - *ticket-id*

### CODE REVIEW

File: src/bindings/c/src/ov_infer_request.cpp [hunk 1]
```diff
@@ -295,7 +295,7 @@ ov_status_e ov_infer_request_wait_for(ov_infer_request_t* infer_request, const i
     }
     CATCH_OV_EXCEPTIONS
 
-    return ret ? ov_status_e::OK : ov_status_e::UNEXPECTED;
+    return ret ? ov_status_e::OK : ov_status_e::RESULT_NOT_READY;
```
Comment (vurusovs) [inline src/bindings/c/src/ov_infer_request.cpp:298]: I suppose the change should be ported to 2023.3

@ilya-lavrenov

**Review by vurusovs — APPROVED** [review]

File: src/bindings/c/docs/api_overview.md [hunk 2]
```diff
@@ -449,7 +449,7 @@ This strcut represents OpenVINO entity and allows you to manipulate with plugins
 
 - `ov_status_e ov_core_read_model(const ov_core_t* core, const char* model_path, const char* bin_path, ov_model_t** model)`
 
-  - Description: Reads models from IR / ONNX / PDPD / TF / TFLite formats to create ov_model_t.
+  - Description: Reads models from IR / PyTorch / ONNX / PDPD / TF / TFLite formats to create ov_model_t.
```
Comment (ilya-lavrenov) [inline src/bindings/c/docs/api_overview.md:452]: we don't support reading of PyTorch formats. We can only convert model (Python object) from PyTorch via convert_model

File: src/bindings/c/README.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ OpenVINO C API has the following structure:
  * [src](./src) contains the implementations of all C APIs.
  * [tests](./tests) contains all tests for OpenVINO C APIs. [Learn more](./docs/how_to_write_unit_test.md).
 
-> **NOTE**:  Using API 2.0 is strongly recommended. Legacy API (for C) has been removed from 2024.0. 
+> **NOTE**: Legacy Inference Engine API (for C) has been removed from 2024.0.
```
Comment (ilya-lavrenov) [inline src/bindings/c/README.md:32]: can we drop info about old API? 
The fact it's removed, is not very useful for new users :)

File: src/bindings/c/src/ov_infer_request.cpp (hunk 1 above)
Comment (ilya-lavrenov) [reply src/bindings/c/src/ov_infer_request.cpp:298]: agree

**Review by zhaixuejun1993 — APPROVED** [review]

File: src/bindings/c/docs/api_overview.md (hunk 2 above)
Comment (riverlijunjie) [reply src/bindings/c/docs/api_overview.md:452]: ok, remove it.

File: src/bindings/c/README.md (hunk 3 above)
Comment (riverlijunjie) [reply src/bindings/c/README.md:32]: agree, will remove it.

File: src/bindings/c/docs/api_overview.md (hunk 2 above)
Comment (vurusovs) [reply src/bindings/c/docs/api_overview.md:452]: My bad)

**Review by ilya-lavrenov — APPROVED** [review]

**Pull request merged**